### DISCUSSION TITLE
Github release, release commit and tag chicken and egg problem

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#1489] [Q&A]
Hi,

In my current jreleaser config, the release process will :

- create a github release
- append the Changelog to a file in the repo

I also generate a commit separately for the release, which contains the new computed version and the appended Changelog.

I have the problem that jreleaser will create the release in github before the new commit is created, and github expects a tag for the release.

It means that the tag is on the wrong commit.

I've been scratching my head on how to solve this, but haven't found something that would work nicely.

Would anyone have any ideas?

Thanks

**Comment by aalmiray** [comment]
I usually make the commit first, then run the release/tag operation.

**Reply by gotson** [reply]
My commit need the append Changelog. I think by running jreleaserChangelog it would generate the append, and then I can commit, then perform the jreleaser release.
